Build failure: pmount

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`nix build github:NixOS/nixpkgs/9abb87b552b7f55ac8916b6fc9e5cb486656a2f3#legacyPackages.x86_64-linux.pmount`


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/282775271

### Relevant log output

```console
make[2]: Entering directory '/build/pmount-0.9.23/src'
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c fs.c
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -c luks.c
luks.c: In function 'luks_create_lockfile':
luks.c:147:7: error: implicit declaration of function 'creat' [-Wimplicit-function-declaration]
  147 |   f = creat( path, 0600);
      |       ^~~~~
luks.c:153:3: error: implicit declaration of function 'close'; did you mean 'pclose'? [-Wimplicit-function-declaration]
  153 |   close(f);
      |   ^~~~~
      |   pclose
luks.c: In function 'luks_remove_lockfile':
luks.c:183:5: error: implicit declaration of function 'unlink' [-Wimplicit-function-declaration]
  183 |     unlink(path);
      |     ^~~~~~
make[2]: *** [Makefile:378: luks.o] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `""`
 - channels(slava): `""`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`


### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pm/pmount/package.nix
+++ b/pkgs/by-name/pm/pmount/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  fetchpatch,
   intltool,
   ntfs3g,
   util-linux,
@@ -21,6 +22,35 @@ stdenv.mkDerivation rec {
     url = "mirror://debian/pool/main/p/pmount/pmount_${version}.orig.tar.bz2";
     sha256 = "db38fc290b710e8e9e9d442da2fb627d41e13b3ee80326c15cc2595ba00ea036";
   };
+
+  patches =
+    let
+      # https://salsa.debian.org/debian/pmount/-/tree/debian/master/debian/patches
+      fetchDebPatch =
+        { name, hash }:
+        fetchpatch {
+          inherit name hash;
+          url = "https://salsa.debian.org/debian/pmount/-/raw/ba05283d4a53aba5349d4397a98d9f45206fb29f/debian/patches/${name}";
+        };
+    in
+    map fetchDebPatch [
+      {
+        name = "10_fix-spelling-binary-errors.patch";
+        hash = "sha256-G4GsUe1ZdYB7Qv333X1hUjOELITR8A2pqyfEnMDTwHI=";
+      }
+      {
+        name = "20_fix-spelling-manpage-error.patch";
+        hash = "sha256-9phF8s7MFSjkhPP24cipeBUps5W1L7YmAE0B1QPx5jk=";
+      }
+      {
+        name = "fix-implicit-function-declaration.patch";
+        hash = "sha256-kdwdS9G1X5RtQFKzF6oMIUubGNP7n1ZQNHu8sN1oV4Q=";
+      }
+      {
+        name = "30_exfat-support.patch";
+        hash = "sha256-kg9gLhOtdrEDlZfUnT910xI5rNR1zgKKRx2kvFQjbi8=";
+      }
+    ];
 
   nativeBuildInputs = [
     intltool

